lib/zendesk_api/middleware/request/retry.rb

```diff
@@ -8,19 +8,32 @@ module Request
       # @private
       class Retry < Faraday::Middleware
         DEFAULT_RETRY_AFTER = 10
-        ERROR_CODES = [429, 503]
+        DEFAULT_ERROR_CODES = [429, 503]
 
         def initialize(app, options = {})
           super(app)
           @logger = options[:logger]
+          @error_codes = options.key?(:retry_codes) && options[:retry_codes] ? options[:retry_codes] : DEFAULT_ERROR_CODES
         end
 
         def call(env)
           original_env = env.dup
-          response = @app.call(env)
+          exception_happened = false
+          begin
```

Comment:
This is somehow, changing the default behaviour of the library. And since this library is used in a lot of external and internal projects, I'd prefer to avoid changing any default logic if we can avoid it. For instance, some applications might be expecting this call to raise an exception, and we are changing it.

Instead, I'd suggest adding a configuration directive (as in this [PR](https://github.com/zendesk/zendesk_api_client_rb/pull/455/files)). Something like 

```
#   in lib/zendesk_api/configuration.rb
config.retry_on_exceptions = false
```

Then, in the actual retry file, you could have something like this (pseudo code)

```
begin
  @app.call(env)
  retry! if @error_codes.include?(response.env[:status])
 rescue StandardError => e
   retry! if config.retry_on_exceptions
```

Of course, you would need to enable this config option in you application.

Reply:
will add config, was thinking about that as well as the other configurable items.  Adding now.

Reply:
fixed!